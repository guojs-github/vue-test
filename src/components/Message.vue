<template>
	<div id="message-list">
		<MessageList :messages="messages"/>
	</div>
</template>

<script>
import MessageList from './MessageList'

export default {
	name: 'Message',
	data: () => ({ messages: ['Hey John', 'Howdy Paco'] }),
	components: {
		MessageList
	}
}
</script>