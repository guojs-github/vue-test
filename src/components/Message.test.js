import Vue from 'vue';
import Message from "./Message";

describe("Message.test.js", () => {
	let cmp, vm;

	beforeEach(() => {
		cmp = Vue.extend(Message); // Create a copy of the original component
		vm = new cmp({
			data: {
				// Replace data value with this fake data
				messages: ["Cat"]
			}
		}).$mount(); // Instances and mounts the component
	});

	it('equals messages to ["Cat"]', () => { // 检验组件数据是否符合要求
		expect(vm.messages).toEqual(["Cat"]);
	});

	it("has the expected html structure", () => { // 检验html显示结构是否符合要求
		expect(vm.$el).toMatchSnapshot();
	});	
});