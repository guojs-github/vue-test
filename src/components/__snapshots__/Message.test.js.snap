// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message.test.js has the expected html structure 1`] = `
<div
  id="message-list"
>
  <ul>
    <li>
      
		Cat
	
    </li>
  </ul>
</div>
`;
